Update docs around version discovery (#602)
twisted/towncrier: docs/tutorial.rst

> Detecting Dates & Versions

Reviewer: Maybe it's best to have a separate section for version and date.

Most of the info from this section is about version.

A separate section should make it easier to jump to the information aboute date.

and I think that it only detects the current version, so not plural.

```suggestion
Detecting version
```

--------------------------

> ``towncrier`` needs to know what version your project is. These are the ways you can provide it (and their order of precedence):

Reviewer: I think it's best to use [Semantic newlines](https://sembr.org/) for documentation content.

Maybe also update this page
https://towncrier.readthedocs.io/en/latest/contributing.html

I guess that not every operaition needs the version.
I don't expect to requiere the version for a `create` command.

It's not 100% clear what's the order of precendece.

```suggestion
Most ``towncrier`` commands, like `build` or `check`, needs to know what version your project is.
These are the ways you can provide it and their order of precedence.
A lower precedence number will overwrite any other method with a higher number:
```

1. Manually pass ``--version=<myversionhere>`` when interacting with ``towncrier``.

> 2. Set a ``version`` key in your configuration file.

Reviewer: ```suggestion
2. Set a value for the ``version`` option in your configuration file.
```
